### PR TITLE
Adds better error management on pipeline/preview

### PR DESCRIPTION
**Pull request by jLopezbarb**
Signed-off-by: Javier López Barba <javier@okteto.com>

Fixes #

## Proposed changes

- Adds better error management on pipeline and preview commands 
-

### CODE REVIEW

File: cmd/pipeline/deploy.go [hunk 1]
```diff
@@ -220,35 +220,39 @@ func waitToBeDeployed(ctx context.Context, name, namespace string, timeout time.
...
+			for name, status := range resourceStatus {
+				if status != "running" {
+					areAllRunning = false
+				}
+				if status == "error" {
+					errors[name] = 1
+				}
+			}
+			if len(errors) > 0 {
+				return fmt.Errorf("pipeline '%s' deployed with resource errors", name)
```
Comment (rberrelleza) [inline cmd/pipeline/deploy.go:249]:
```suggestion
				return fmt.Errorf("pipeline '%s' deployed with errors", name)
```

File: cmd/preview/deploy.go [hunk 2]
```diff
@@ -233,33 +233,37 @@ func waitToBeDeployed(ctx context.Context, name, namespace string, timeout time.
...
+			for name, status := range resourceStatus {
+				if status != "running" {
+					areAllRunning = false
+				}
+				if status == "error" {
+					errors[name] = 1
+				}
+			}
+			if len(errors) > 0 {
+				return fmt.Errorf("preview environment '%s' deployed with resource errors", name)
```
Comment (rberrelleza) [inline cmd/preview/deploy.go:262]:
```suggestion
				return fmt.Errorf("preview environment '%s' deployed with errors", name)
```

Comment (rberrelleza) [reply cmd/preview/deploy.go:262]: do you think it would be helpful if we added a hint with the URL of the preview env?

**Review by ifbyol — APPROVED** [review]

**Review by pchico83 — APPROVED** [review]

File: cmd/pipeline/deploy.go [hunk 3]
```diff
@@ -220,35 +220,48 @@ func waitToBeDeployed(ctx context.Context, name, namespace string, timeout time.
...
 	}
-	return nil
+}
+
+func getPreviewEnvURL(ctx context.Context, name string) string {
+	url := okteto.GetURL()
+	if url == "na" {
+		return ""
+	}
+	return fmt.Sprintf("https://cloud.okteto.com/#/previews/%s", name)
```
Comment (pchico83) [inline cmd/pipeline/deploy.go:264]:
```suggestion
	return fmt.Sprintf("https://%s/#/previews/%s", url , name)
```

File: cmd/pipeline/deploy.go [hunk 4]
```diff
@@ -220,35 +220,48 @@ func waitToBeDeployed(ctx context.Context, name, namespace string, timeout time.
...
+					areAllRunning = false
+				}
+				if status == "error" {
+					errors[name] = 1
+				}
 			}
-			if status == "error" {
-				errors[name] = 1
+			if len(errors) > 0 {
+				previewEnvURL := getPreviewEnvURL(ctx, name)
```
Comment (pchico83) [inline cmd/pipeline/deploy.go:249]: `getNamespaceURL`? This should be the url of the namespace where the pipeline is being executed

File: cmd/pipeline/deploy.go [hunk 5]
```diff
@@ -220,35 +220,48 @@ func waitToBeDeployed(ctx context.Context, name, namespace string, timeout time.
...
-		for name, status := range resourceStatus {
-			if status != "running" {
-				areAllRunning = false
+
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("pipeline '%s' didn't finish after %s", name, timeout.String())
+		case <-ticker.C:
+			resourceStatus, err := okteto.GetResourcesStatusFromPipeline(ctx, name, namespace)
```
Comment (pchico83) [inline cmd/pipeline/deploy.go:235]: This method is wrong. It should query the deployments and statefulsets of the pipeline, not of the namespace:
https://github.com/okteto/okteto/blob/master/pkg/okteto/pipeline.go#L192

File: cmd/preview/deploy.go [hunk 6]
```diff
@@ -233,33 +233,37 @@ func waitToBeDeployed(ctx context.Context, name, namespace string, timeout time.
...
-		for name, status := range resourceStatus {
-			if status != "running" {
-				areAllRunning = false
+
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("preview environment '%s' didn't finish after %s", name, timeout.String())
+		case <-ticker.C:
+			resourceStatus, err := okteto.GetResourcesStatusFromPreview(ctx, namespace)
```
Comment (pchico83) [inline cmd/preview/deploy.go:248]: This method is wrong, it should check the status of individual deployments/statefulsets, but also if they belong to a gitDeploy, app or stack:
https://github.com/okteto/okteto/blob/master/pkg/okteto/pipeline.go#L192

**Review by pchico83 — APPROVED** [review]

File: cmd/pipeline/deploy.go (hunk 5 above)
Comment (jLopezbarb) [reply cmd/pipeline/deploy.go:235]: It's already filtered

**Pull request merged**